### PR TITLE
#303 Frontend/linking dataset from search results to a dataset view page

### PR DESCRIPTION
**Pull request by Vatika1**
This current branch allows the user to choose a specific dataset from search results page and view its full contents in uneditable mode.

https://user-images.githubusercontent.com/35615793/107463296-c357a980-6b2b-11eb-9084-e259653b65e8.mp4

### CODE REVIEW

File: Client/src/Components/DatasetUpload/DatasetUploadView.tsx [hunk 1]
```diff
@@ -26,10 +33,27 @@ const fixPartialForform = (partialDataset: Partial<IDatasetModel>): IDatasetMode
...
 export const DatasetUploadView = (props: IProps) => {
+  const { datasetID } = useParams<IDatasetViewParams>()
+
   const location = useLocation()
   const initialSentDataset = location.state as IDatasetModel
-  let initialValues = props.initialDataset || initialSentDataset || defaultDatasetModel
-  initialValues = fixPartialForform(initialValues)
+  const [initialValues, setInitialValues] = useState(props.initialDataset || initialSentDataset || defaultDatasetModel)
+  const [editable, setEditable] = useState(true)
+  //initialValues = fixPartialForform(initialValues)
```
Comment (GD-Dheer) [inline Client/src/Components/DatasetUpload/DatasetUploadView.tsx:42]: Comment

File: Client/src/Components/DatasetUpload/DatasetUploadView.tsx [hunk 2]
```diff
@@ -26,10 +33,27 @@ const fixPartialForform = (partialDataset: Partial<IDatasetModel>): IDatasetMode
...
+
+  useEffect(() => {
+    const getDatasetInfo = async (id: number) => {
+      const datasetArray = await callGetDatasets({ datasetId: [id] })
+      const dataset = datasetArray[0]
+      setInitialValues(fixPartialForform(dataset))
+      setEditable(false)
+      console.log(dataset)
```
Comment (GD-Dheer) [inline Client/src/Components/DatasetUpload/DatasetUploadView.tsx:50]: Log

File: Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx [hunk 3]
```diff
@@ -28,9 +28,9 @@ const referenceValidationSchema = Yup.object().shape({
 
 export const variableValidationSchema = Yup.object().shape(
   {
-    name: Yup.string().trim().required(requiredMessage('Name')),
-    repr: Yup.string().trim().required(requiredMessage('Representation')),
-    units: Yup.string().trim().required(requiredMessage('Units'))
+    name: Yup.string().trim(),
```
Comment (GD-Dheer) [inline Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx:31]: You can remove this validation schema. Were not gonna use this formatting anymore

File: Client/src/Components/Search/SearchResults.tsx [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
+import React, { useState } from 'react'
+import { DatasetFormModal } from '../DatasetUpload/DatasetViewModal'
 
 interface IProps {
   datasetResults: IDatasetModel[],
   handleSelectionChanged: (selection: SelectionChangeParams) => void,
   button?: any,
 }
 
+interface IProps2 {
```
Comment (GD-Dheer) [inline Client/src/Components/Search/SearchResults.tsx:13]: Update this props name

File: Client/src/Components/Search/SearchResults.tsx [hunk 5]
```diff
@@ -31,8 +35,17 @@ export const SearchResults = (props: IProps) => {
     return `${reference.year}`
   }
 
+  const [open, setOpen] = useState(false)
+
+  const getNameLink = (params: ValueGetterParams) => {
```
Comment (GD-Dheer) [inline Client/src/Components/Search/SearchResults.tsx:40]: ValueGetterParams? Quick name change here

**Review by GD-Dheer — APPROVED** [review]
Small changes to adjust. Other than that LGTM! Thanks for the hard work

**Review by ds161833 — APPROVED** [review]
Ui looks clean good job

File: Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx (hunk 3 above)
Comment (Vatika1) [reply Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx:31]: @GD-Dheer you want me to remove the entire file?

Comment (ds161833) [reply Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx:31]: We're gonna get back to this in the units epic, I asked Leslie to make it like this

File: Client/src/Components/DatasetUpload/DatasetUploadView.tsx (hunk 2 above)
Comment (Vatika1) [reply Client/src/Components/DatasetUpload/DatasetUploadView.tsx:50]: removed

File: Client/src/Components/DatasetUpload/DatasetUploadView.tsx (hunk 1 above)
Comment (Vatika1) [reply Client/src/Components/DatasetUpload/DatasetUploadView.tsx:42]: removed

File: Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx (hunk 3 above)
Comment (ds161833) [reply Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx:31]: So leave this as it is tight now

Comment (silent-po2) [reply Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx:31]: Thanks for this change, while not directly related to the PR itself, this has been a slight headache in testing

**Review by silent-po2 — APPROVED** [review]
Solid work

File: Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx (hunk 3 above)
Comment (Vatika1) [reply Client/src/Components/DatasetUpload/DatasetValidationSchema.tsx:31]: yeah i will merge without deleting this

File: Client/src/Components/Search/SearchResults.tsx (hunk 4 above)
Comment (Vatika1) [reply Client/src/Components/Search/SearchResults.tsx:13]: removed it

**Pull request merged**